### ISSUE TITLE
Lifecycle API

### ISSUE THREAD

**Issue by siusin** [w3c/strategy#92]

**Comment by spanicker** [comment]
Here's a public version of the slides:
https://docs.google.com/presentation/d/1nBZKbdCDAfMFhQ3Fkkc6JjliC5TjknMKCpU1ZiVKgvY/edit#slide=id.p
Could you update the link?

**Comment by siusin** [comment]
I've updated the link in this issues and the meeting minutes. Thanks @spanicker , it's great to have a public version.

**Comment by siusin** [comment]
@spanicker 
the [WebPlatform WG](http://w3c.github.io/charter-html/webplat-wg.html#scope)  said they would be delighted to accept this proposal if you think WebPlat is a better fit. Please send a request to team-webplatform@w3.org (or ping me) once you think this API is ready-to-migrate. Thanks!

**Comment by tidoust** [comment]
@siusin, regarding "Does this API have any overlaps with our past/future work?", I note previous attempts to discuss application lifecycle in the now dead [Systems App WG](http://www.w3.org/2012/sysapps/). These attemps were very different in scope in practice because the WG was trying to develop OS-level APIs, not browser APIs. For instance, if I understand the proposal correctly, it does not attempt to expose an API that would let the application control its lifecycle, the lifecycle being controlled by the OS (or the browser) at all times.

The following specs might still be worth checking for inspiration:

- The [Application Lifecycle and Events](http://rawgit.com/sysapps/app-lifecycle/gh-pages/index.html) spec defined APIs for managing the lifecycle of an application and associated events (note URL uses rawgit because sysapps domain no longer seems to exist, which breaks the Table of Contents and actually breaks the spec in Edge, oh well).
- The [Runtime and Security Model for Web Applications](http://www.w3.org/TR/runtime/) spec defined an [application life cycle](http://www.w3.org/TR/runtime/#application-life-cycle).
- The group was also supposed to work on an Idle API to report events when the user goes idle, inspired by the [Boot2Gecko Idle API](https://wiki.mozilla.org/WebAPI/IdleAPI).

The term "application lifecycle" also often appears in discussions related to interactive TV (iTV) specifications (ATSC, HbbTV, Hybridcast) for applications that get streamed with a broadcast channel. There again, the notion of lifecycle is probably slightly different, but needs may still be similar. I would expect discussions on application lifecycle to be in scope of the [Media and Entertainment IG](https://www.w3.org/2011/webtv).

**Comment by spanicker** [comment]
Thanks for the links!
There is some overlap with the linked specs. As you indicated current proposal does not let app control its lifecycle, it is always controlled by the browser.
Any idea if these linked specs are being pursued or any ongoing work here? Should I reach out to the spec authors?

**Comment by tidoust** [comment]
@spanicker I am not aware of any on-going work on the specs I linked although some of the features in these specs may have ended in other specs (I don't think that's the case for application cycle features though).

You can certainly reach out to Anssi (@anssiko) and Mounir (@mounirlamouri) (who now works for Google) to gather feedback and exchange ideas.

**Comment by tidoust** [comment]
I note that the proposal is now being incubated in the Web Platform Incubator Community Group (WICG) under the name [A lifecycle for the Web](https://github.com/WICG/web-lifecycle)

**Comment by siusin** [comment]
Right, the WebPerf WG had a very good [discussion](https://www.w3.org/2017/11/06-webperf-minutes.html#item03) during TPAC 2017.

**Comment by svgeesus** [comment]
Looks like this activity stalled three years ago. Did it move elsewhere?

**Comment by svgeesus** [comment]
Looks like this activity stalled three years ago. Did it move elsewhere?
